students/rhamersky/Lesson_10/mailroom_part6.py

```diff
@@ -0,0 +1,392 @@
+        '''
+        # Creates a donor name list
+        # lst_donor_name = [dic_donors[row][0] for row in range(len(dic_donors))]  # --> Added list comprehension
+        # Checks to see if donor has given before and updates history
+        dic_new_donor_names = dic_donors
+        if donor_name in dic_donors:
+            dic_new_donor_names[donor_name][1] = dic_donors[donor_name][
+                                                     1] + 1  # --> Increase amount of times donor donated
+            dic_new_donor_names[donor_name][0] = dic_donors[donor_name][
+                                                     0] + donation_amount  # --> Adds donation to history
+            lst_donor_data = dic_new_donor_names[donor_name]  # --> Creates a list of the values in the dictionary.
+            lst_donor_data.append(donation_amount)  # --> Appends donation amount to donor data list
+            dic_new_donor_names[donor_name] = lst_donor_data  # --> Replaces the old value list with lst_donor_data.
+        # Adds new donor to the donor list
+        else:
+            dic_new_donor_names[donor_name] = [donation_amount, initial_donation, donation_amount]
+        return dic_new_donor_names
+
+    @staticmethod
+    def print_donor_names():
+        '''
+        Prints donors names if the user types "list" when the program prompts the user to input a donor.
+        :return:
+        '''
+        # Creates a list just of donor names
+        print("\n" + "Donor")
+        for name in dic_donors:
+            print(name)
+
+    @staticmethod
+    def dic_sort(dic_donors):
+        '''
+        Takes a dictionary and creates a sorted tuple using the first element of the value list.
+        It then puts the sorted data into a new dictionary.
+        :param dic_donors: The dictionary of donors.
+        :return:
+        '''
+        dic_donors_total_donation = {}
+        # Removes all past donations from the value list.
+        for key in dic_donors:
+            lst_total_donation = dic_donors[key]
+            dic_donors_total_donation[key] = lst_total_donation
+        sorted_tuple = sorted(dic_donors_total_donation.items(), key=lambda x: x[1], reverse=True)
+        sorted_dic = {}
+        for items in sorted_tuple:
+            sorted_dic[items[0]] = [items[1][0], items[1][1]]
+        return sorted_dic
+
+    @staticmethod
+    def create_report(dic_donors):
+        '''
+        This method will create a report for the user with the following columns:
+        donor's name, number of gifts, average gift amount, and lifetime donations.
+        :param dic_donors: The dictionary of donors.
+        :return:
+        '''
+        longest_entry = float()
+        # Finds the max amount of characters for all items stored in dic_donors.
+        for name in dic_donors:
+            entry_length = len(name)
+            if longest_entry < entry_length:
+                longest_entry = entry_length
+            for index in range(0,1):
```

Comment:
Don't need to use a loop with a single iteration, also the first arg of `range()` defaults to `0`